extensions/import/src/wizard/pages/exportConfigPage.ts

```diff
@@ -0,0 +1,270 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+import * as sqlops from 'sqlops';
+import * as nls from 'vscode-nls';
+import * as vscode from 'vscode';
+import * as os from 'os';
+import * as path from 'path';
+import { DacFxDataModel } from '../api/models';
+import { DacFxWizard } from '../dacFxWizard';
+import { DacFxPage } from '../api/dacFxPage';
+
+const localize = nls.loadMessageBundle();
+
+export class ExportConfigPage extends DacFxPage {
+
+	protected readonly wizardPage: sqlops.window.modelviewdialog.WizardPage;
+	protected readonly instance: DacFxWizard;
+	protected readonly model: DacFxDataModel;
+	protected readonly view: sqlops.ModelView;
+
+	private serverDropdown: sqlops.DropDownComponent;
+	private databaseDropdown: sqlops.DropDownComponent;
+	private form: sqlops.FormContainer;
+	private fileTextBox: sqlops.InputBoxComponent;
+	private fileButton: sqlops.ButtonComponent;
+
+	private databaseLoader: sqlops.LoadingComponent;
+
+	public constructor(instance: DacFxWizard, wizardPage: sqlops.window.modelviewdialog.WizardPage, model: DacFxDataModel, view: sqlops.ModelView) {
+		super(instance, wizardPage, model, view);
+	}
+
+	async start(): Promise<boolean> {
+		let databaseComponent = await this.createDatabaseDropdown();
+		let serverComponent = await this.createServerDropdown();
+		let fileBrowserComponent = await this.createFileBrowser();
+
+		this.form = this.view.modelBuilder.formContainer()
+			.withFormItems(
+				[
+					serverComponent,
+					databaseComponent,
+					fileBrowserComponent,
+				], {
+					horizontal: true
+				}).component();
+		await this.view.initializeModel(this.form);
+		return true;
+	}
+
+	async onPageEnter(): Promise<boolean> {
+		let r1 = await this.populateServerDropdown();
+		let r2 = await this.populateDatabaseDropdown();
+		return r1 && r2;
+	}
+
+	async onPageLeave(): Promise<boolean> {
+		return true;
+	}
+
+	public async cleanup(): Promise<boolean> {
+		return true;
+	}
+
+	public setupNavigationValidator() {
+		this.instance.registerNavigationValidator(() => {
+			if (this.databaseLoader.loading) {
+				return false;
+			}
+			return true;
+		});
+	}
+
+	private async createServerDropdown(): Promise<sqlops.FormComponent> {
```

Comment:
it looks like this function and others are duplicated elsewhere in the code.  Can we reuse the same implementation to avoid this duplication?

Reply:
I moved part of the server and database dropdown population to a new base class and other common methods in the DacFx config pages to another base class.